src/core/settings.c

```diff
@@ -762,21 +762,31 @@ static CONFIG_REC *parse_configfile(const char *fname)
 	return config;
 }
 
-static void init_configfile(void)
+static void init_ensure_dir(const char *dname, int dmode)
 {
 	struct stat statbuf;
-	char *str;
 
-	if (stat(get_irssi_dir(), &statbuf) != 0) {
-		/* ~/.irssi not found, create it. */
-		if (g_mkdir_with_parents(get_irssi_dir(), 0700) != 0) {
-			g_error("Couldn't create %s directory: %s",
-			        get_irssi_dir(), g_strerror(errno));
+	if (stat(dname, &statbuf) != 0) {
+		/* directory not present */
+		if (g_mkdir_with_parents(dname, dmode) != 0) {
+			g_error("Couldn't create %s directory: %s", dname, g_strerror(errno));
 		}
 	} else if (!S_ISDIR(statbuf.st_mode)) {
-		g_error("%s is not a directory.\n"
-			"You should remove it with command: rm %s",
-			get_irssi_dir(), get_irssi_dir());
+		g_error("%s is not a directory. \n"
+		        "You should remove it with command: rm %s",
+		        dname, dname);
+	}
+}
+
+static void init_configfile(void)
+{
+	char *str;
+
+	init_ensure_dir(get_irssi_dir(), 0700); /* XDG_DATA_HOME or ~/.irssi*/
+	if (!strcmp(get_irssi_dir(), get_irssi_cache_dir()) &&
```

Comment:
is this condition also mixed up ?

Reply:
No, we call `init_ensure_dir()` only when `get_irssi_dir()` is different to `get_irssi_cache_dir()` and `get_irssi_runtim_dir()`

Reply:
.... but `!strcmp` means **equal**

maybe write `strcmp.... != 0`?

Reply:
I'm currently having trouble finding time to get back on this issue. If someone would like to make a commit, it would be much appreciated. Sorry! :cry:

Reply:
anyway thank you for the contribution I think you got pretty far